File: dagger/app/src/main/java/com/dagger/mvp/presenter/MVPPresenterImpl.java
package com.dagger.mvp.presenter;

import com.dagger.mvp.MVPApi;
import com.dagger.mvp.model.MVPBean;
import com.dagger.mvp.view.MVPView;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.network.manager.RxNetWork;
import io.reactivex.network.manager.RxNetWorkListener;

/**
 * by y on 2017/5/31.
 */
public class MVPPresenterImpl implements MVPPresenter {

    private final MVPView mvpView;

    @Inject
    public MVPPresenterImpl(MVPView mvpView) {
        this.mvpView = mvpView;
    }


    @Override
    public void startNetWork() {
        RxNetWork.getInstance().cancel(RxNetWork.TAG);
        RxNetWork
                .getInstance()
                .setBaseUrl(MVPApi.ZL_BASE_API)
                .getApi(RxNetWork.observable(
                        MVPApi.ZLService.class).getList("daily", 20, 0),
                        new RxNetWorkListener<List<MVPBean>>() {
                            @Override
                            public void onNetWorkStart() {
                                mvpView.showProgress();
                            }

                            @Override
                            public void onNetWorkError(Throwable e) {
                                mvpView.onNetError();
                                mvpView.hideProgress();
                            }

                            @Override
                            public void onNetWorkComplete() {
                                mvpView.hideProgress();
                            }

                            @Override
                            public void onNetWorkSuccess(List<MVPBean> data) {
                                mvpView.onNetSuccess(data);
                            }
                        });
    }
}

File: UtilsSample/Dagger/src/main/java/sample/util/develop/android/dagger/mvp/presenter/MVPPresenterImpl.kt
package sample.util.develop.android.dagger.mvp.presenter


import io.reactivex.network.RxNetWork
import io.reactivex.network.RxNetWorkListener
import sample.util.develop.android.dagger.mvp.MVPApi
import sample.util.develop.android.dagger.mvp.model.MVPBean
import sample.util.develop.android.dagger.mvp.view.MVPView

import javax.inject.Inject

/**
 * by y on 2017/5/31.
 */
class MVPPresenterImpl @Inject
constructor(private val mvpView: MVPView) : MVPPresenter {

    override fun startNetWork() {
        RxNetWork.instance.cancelDefaultKey()
        RxNetWork
            .instance
            .getApi(RxNetWork.observable(MVPApi.ZLService::class.java).getList("daily", 20, 0),
                object : RxNetWorkListener<List<MVPBean>> {
                    override fun onNetWorkStart() {
                        mvpView.showProgress()
                    }

                    override fun onNetWorkError(e: Throwable) {
                        mvpView.onNetError()
                        mvpView.hideProgress()
                    }

                    override fun onNetWorkComplete() {
                        mvpView.hideProgress()
                    }

                    override fun onNetWorkSuccess(data: List<MVPBean>) {
                        mvpView.onNetSuccess(data)
                    }
                })
    }
}
